python/ray/data/datasource/file_based_datasource.py

```diff
@@ -497,11 +502,24 @@ def _resolve_paths_and_filesystem(
                     _encode_url(path), filesystem
                 )
                 resolved_path = _decode_url(resolved_path)
+            elif "Unrecognized filesystem type in URI" in str(e):
+                scheme = urllib.parse.urlparse(path, allow_fragments=False).scheme
+                if scheme in ["http", "https"]:
+                    # If scheme of path is HTTP and filesystem is not resolved,
+                    # try to use fsspec HTTPFileSystem. This expects fsspec is
+                    # installed.
+                    from fsspec.implementations.http import HTTPFileSystem
```

Comment:
can you check if fsspec is installed before calling this import?

Reply:
@richardliaw - make sense, updated, also added a clearer error message if import is failed.